contrib/win32/win32compat/w32fd.c

```diff
@@ -1154,13 +1064,19 @@ spawn_child_internal(char* cmd, char *const argv[], HANDLE in, HANDLE out, HANDL
 	si.hStdOutput = out;
 	si.hStdError = err;
 	si.dwFlags = STARTF_USESTDHANDLES;
-
-	debug3("spawning %ls", cmdline_utf16);
 	
-	if (as_user)
-		b = CreateProcessAsUserW(as_user, NULL, cmdline_utf16, NULL, NULL, TRUE, flags, NULL, NULL, &si, &pi);
-	else
-		b = CreateProcessW(NULL, cmdline_utf16, NULL, NULL, TRUE, flags, NULL, NULL, &si, &pi);
+	int num = 0;
+	char * t;
+	do {
+		t = cmdline_utf16 + num++;
+		debug3("spawning %ls", t);
+		if (as_user)
+			b = CreateProcessAsUserW(as_user, NULL, t, NULL, NULL, TRUE, flags, NULL, NULL, &si, &pi);
+		else
+			b = CreateProcessW(NULL, t, NULL, NULL, TRUE, flags, NULL, NULL, &si, &pi);
+		*(cmdline_utf16 + wcslen(cmdline_utf16) - 1) = L'\0';
+		/*failover only when double quotes is potentially added necessary*/
```

Comment:
you could simplify the below condition by putting this inside loop and getting rid of "num" and "t":

if  ( !b && GetLastError() != ERROR_FILENOTFOUND && (argv == NULL || *argv == NULL) && cmd[0] != '\"') {
        cmdline_utf16++;
        *(cmdline_utf16 + wcslen(cmdline_utf16) - 1) = L'\0';
}

break;

Reply:
I can get rid of num, but not t.
I need to free cmdline_utf16 at the end. the pointer can't change

Reply:
I can't get rid of num since I need to make sure only one more time is tried